tests/Feature/TrackerSpaNavigationTest.php

```
@@ -153,3 +153,35 @@ function stepEngagement( array $report, string $label ): array
 	}
 } );
+
+test( 'a page view is on the wire before the engagement update that refers to it', function (): void {
+	// Page views sit in the batch queue; engagement updates go out immediately.
+	// Run at the production batch interval rather than the harness's short one,
+	// which is fast enough to hide the race entirely.
+	$report = runTrackerHarness( [ 'batchInterval' => 5000 ] );
+
+	$seenPages = [];
+
+	foreach ( $report['timeline'] as $beacon ) {
```

PHPStan (2.2.7)

[error] Offset 'timeline' does not exist on array{initialPageViews: list<string>, steps: list<array{label: string, pageViewPaths: list<string>, engagementUpdates: list<array{path: string|null, scroll_depth: int|null}>}>}.


(offsetAccess.notFound)